### PR TITLE
ES Producer Hook tests rewrite

### LINKED ISSUE #271: Rewrite ES event hook/producer tests
In PR #266, I deleted the old [producer tests](https://github.com/threatgrid/ctia/blob/e138cb9aa848d7c2a6ff9401ae5260008e01b8db/test/ctia/events/producer_test.clj), because I deleted the old producer implementation.

Need to write new tests as a sibling to the [redis event hook tests](https://github.com/polygloton/ctia/blob/issue-255-part1b/test/ctia/flows/hooks/redis_event_hook_test.clj).  The new ES hook tests would need to trigger the hook as a side effect, as is done for the redis event hook tests, and then query ES to find the events are there.

### PR DESCRIPTION
**Pull request by gbuisson**
closes #271 

- Tests the ES producer hook on both slicing strategies
- also had to change test selectors, see my comment for more details.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -54,15 +54,11 @@
...
                    :disabled :disabled
-                   :es-producer #{:es-producer
-                                   :es-producer-filtered-alias
-                                   :es-producer-aliased-index}
                    :default #(not= :disabled %)
-                   :integration #{:es-store
-                                   :integration
-                                   :es-producer
-                                   :es-producer-filtered-alias
-                                   :es-producer-aliased-index}
```
Comment (gbuisson) [inline project.clj]: this selector syntax was not working, using `lein test :integration` launched 0 tests so I reverted back to use a fn instead. tell me if I got it wrong.

**Comment by yogsototh** [comment]
LGTM

**Comment by polygloton** [comment]
It says rewrite, but nothing was deleted.  Is it meant to be just new tests?

**Comment by gbuisson** [comment]
Tests were deleted with PR #266 you added a task to implement them again.

**Pull request merged**